File: twosum.java
import java.util.*;

public class Solution {

    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int need = target - nums[i];

            if (map.containsKey(need)) {
                return new int[]{map.get(need), i};
            }
            map.put(nums[i], i);
        }
        return new int[]{-1, -1};
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // input size
        int n = sc.nextInt();
        int[] nums = new int[n];

        // input array
        for (int i = 0; i < n; i++) {
            nums[i] = sc.nextInt();
        }

        // input target
        int target = sc.nextInt();

        int[] result = twoSum(nums, target);

        // output
        System.out.println(result[0] + " " + result[1]);
    }
}

File: TwoSum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

class Solution {
public:
    auto twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> map;

        for (int i = 0; i < nums.size(); i++) {
            int need = target - nums[i];

            if (map.find(need) != map.end()) {
                return std::vector<int>{map[need], i};
            }
            map[nums[i]] = i;
        }
        
        return std::vector<int>{-1, -1};
    }
};

int main() {
    int size;
    std::cout << "Enter the number of elements in the array: ";
    std::cin >> size;

    std::vector<int> nums(size);
    std::cout << "Enter " << size << " numbers: ";
    for (int i = 0; i < size; i++) {
        std::cin >> nums[i];
    }

    int target;
    std::cout << "Enter the target sum: ";
    std::cin >> target;

    Solution solution;
    auto result = solution.twoSum(nums, target);

    std::cout << "Indices: [" << result[0] << ", " << result[1] << "]" << std::endl;

    return 0;
}